src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -224,15 +224,10 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                         if (restoreDataObject != null)
                         {
                             // create task metadata
-                            TaskMetadata metadata = new TaskMetadata();
-                            metadata.ServerName = connInfo.ConnectionDetails.ServerName;
-                            metadata.DatabaseName = restoreParams.TargetDatabaseName;
-                            metadata.Name = SR.RestoreTaskName;
-                            metadata.IsCancelable = true;
-                            metadata.Data = restoreDataObject;
+                            TaskMetadata metadata = TaskMetadata.Create(restoreParams, SR.RestoreTaskName, restoreDataObject, ConnectionServiceInstance);
```

Comment:
I removed it from here and if the cancel function is defined  I assume it's cancelable 

---
In reply to: [134042668](https://github.com/Microsoft/sqltoolsservice/pull/438#discussion_r134042668) [](ancestors = 134042668)